### PR TITLE
Metacello cleanup

### PR DESCRIPTION
**Pull request by guillep**
This PR cleans up several things in Metacello code with the objective of easily changing the way code is loaded.

 - remove ambiguities between specs and directives. specs are loaded, then transformed in directives.
   - a record operation works on specs and generates directives
   - a fetch/load operation works on directives and loads files/code
 - remove duplicated code when loading packages
 - inline methods and remove unnecessary double dispatch to largely simplify code

Also, sneak in some fixes in circular dependencies between unordered and sequenceable collections

I tested that the following projects work
- mutalk
- xml parser
- kbg atomic
- seaside
- geometry
- artefact

### CODE REVIEW

File: src/Metacello-Core/MetacelloLinearLoadDirective.class.st [hunk 1]
```diff
@@ -6,21 +6,29 @@ Class {
...
 }
 
-{ #category : 'actions' }
-MetacelloLinearLoadDirective >> loadUsing: aLoaderDirective gofer: aGofer [
+{ #category : 'as yet unclassified' }
+MetacelloLinearLoadDirective >> executeUsing: aLoadingStrategy [
+
+	self loadDirectives isEmpty ifTrue: [ ^ self ].
+
+	(aLoadingStrategy isKindOf: MetacelloAtomicLoadingStrategy) ifTrue: [
```
Comment (guillep) [inline src/Metacello-Core/MetacelloLinearLoadDirective.class.st]: I need to deal with these isKindOf: before this is merged!

File: src/Metacello-Core/MetacelloAtomicLoadDirective.class.st [hunk 2]
```diff
@@ -11,85 +11,31 @@ Class {
...
-	super loadLinearLoadDirective: aLoaderDirective gofer: aGofer
-]
-
-{ #category : 'actions' }
-MetacelloAtomicLoadDirective >> loadPackageDirective: aPackageLoadDirective gofer: aGofer [
-	"accumulate packages"
-	
-	self packageloads add: aPackageLoadDirective
-]
+{ #category : 'as yet unclassified' }
```
Comment (jecisc) [inline src/Metacello-Core/MetacelloAtomicLoadDirective.class.st]: This will trigger the release test

**Review by jecisc — COMMENTED** [review]
The renaming is really nice to differentiate both thanks

**Comment by Ducasse** [comment]
```
SystemNotification: Loading System-Platforms-tonel.1.mcz
An error happened while reading MCZ. We will fallback to another format. Short error stack: ReadStream(Object)>>doesNotUnderstand: #nextWordsInto:
WideString class>>newFromStream:
MCDataStream>>readWordLike
MCDataStream>>ne
```

**Comment by Ducasse** [comment]
@guillep there are failing tests that we can fix.

**Comment by guillep** [comment]
Remaining broken tests are not related to the PR

**Comment by Ducasse** [comment]
I will revert it.

**Pull request merged**